packages/attribute-registry-process/test/db.test.ts

```diff
@@ -0,0 +1,164 @@
+/* eslint-disable functional/no-let */
+import { beforeAll, afterEach, describe, expect, it } from "vitest";
+import { GenericContainer } from "testcontainers";
+import { PostgreSqlContainer } from "@testcontainers/postgresql";
+import {
+  AttributeCollection,
+  ReadModelRepository,
+  initDB,
+} from "pagopa-interop-commons";
+import { v4 as uuidv4 } from "uuid";
+import { Document } from "mongodb";
+import { IDatabase } from "pg-promise";
+import { config } from "../src/utilities/config.js";
+import { AttributeRegistryService } from "../src/services/attributeRegistryService.js";
+import { ReadModelService } from "../src/services/readModelService.js";
+
+describe("database test", () => {
+  let attributes: AttributeCollection;
+  let readModelService: ReadModelService;
+  let attributeRegistryService: AttributeRegistryService;
+  let postgresDB: IDatabase<unknown>;
+  beforeAll(async () => {
+    const postgreSqlContainer = await new PostgreSqlContainer("postgres:14")
+      .withUsername(config.eventStoreDbUsername)
+      .withPassword(config.eventStoreDbPassword)
+      .withDatabase(config.eventStoreDbName)
+      .withCopyFilesToContainer([
+        {
+          source: "../../docker/event-store-init.sql",
+          target: "/docker-entrypoint-initdb.d/01-init.sql",
+        },
+      ])
+      .withExposedPorts(5432)
+      .start();
+
+    const mongodbContainer = await new GenericContainer("mongo:6.0.7")
+      .withEnvironment({
+        MONGO_INITDB_DATABASE: config.readModelDbName,
+        MONGO_INITDB_ROOT_USERNAME: config.readModelDbUsername,
+        MONGO_INITDB_ROOT_PASSWORD: config.readModelDbPassword,
+      })
+      .withExposedPorts(27017)
+      .start();
+
+    config.eventStoreDbPort = postgreSqlContainer.getMappedPort(5432);
+    config.readModelDbPort = mongodbContainer.getMappedPort(27017);
+    attributes = ReadModelRepository.init(config).attributes;
```

Comment:
probably it's better to init the connection just once under `describe` scope and use the obtained collections references

Reply:
As far as I understand, that's exactly what it does. The `beforeAll` includes the setup for the entire suite and it's execute only once (this is not a `beforeEach`).

Reply:
yes, but [here](https://github.com/pagopa/interop-be-monorepo/pull/105/files#diff-61fce8d52d98ee1a85b8c36e6bf395e2f445f67d99ccf086fab9a84aa97d370dR156) the connection is initialized for each call

Reply:
Fixed, I forgot about that. Thanks